src/app/modules/group/components/associated-activity/associated-activity.component.ts

```diff
@@ -23,42 +26,51 @@ type ActivityId = string;
 })
 export class AssociatedActivityComponent implements OnDestroy, ControlValueAccessor {
 
-  rootActivity: null|{
-    id: ActivityId,
-    name: string|null,
-    path: string,
-  } = null;
+  rootActivity: NoActivity|NewActivity|ExistingActivity = { type: 'no-activity' };
+  rootActivityData: null|{name: string|null, path: string|null} = null;
```

Comment:
I initially merged these two states, but because typeguard functions can't be used in the html file, I had to split them, otherwise I have type errors in the html